docs/admin/cluster-management.md

```diff
@@ -14,47 +14,141 @@ certainly want the docs that go with that version.</h1>
 
-## Updgrading to a different API version
+The upgrade is in-place, meaning that all user pods are kept alive throught this process. `kube-push.sh` has a few functionalities. It can either be used to upgrade all Kubernetes components to the currently built binaries (default option), or to do limited upgrade of either the master machine or a single node. Instead of upgrading to the head, you can also specify the version which you want to roll out.
 
-There is a sequence of steps to upgrade to a new API version.
+If you want to upgrade your whole cluster to, say, version 0.20.2 on a provider supported by `kube-push.sh` you use the command:
+```
+cluster/kube-push.sh v.0.20.2
+```
+In case when want to upgrade only running master to the version 0.20.2, you can use:
+```
+cluster/kube-push.sh -m v0.20.2
+```
+and if you want to upgrade a node `my_node` instead you write:
+```
+cluster/kube-push.sh -n my_node v0.20.2
+``` 
 
-1. Turn on the new api version
-2. Upgrade the cluster's storage to use the new version.
-3. Upgrade all config files. Identify users of the old api version endpoints.
-4. Update existing objects in the storage to new version by running cluster/update-storage-objects.sh
-3. Turn off the old version.
+## Scaling the running cluster
+
+When the Kubelet is given masters address it'll self register in running Kubernetes cluster.
+
+If your cluster runs out short on resources you can easily add more machines to it. If you're using GCE or GKE it is done by resizing Instance Group managing your Nodes. It's done by modifying number of instances on `Compute > Compute Engine > Instance groups > your group > Edit group` [Google Cloud Console page](https://console.developers.google.com) or using gcloud CLI:
+```
+gcloud preview managed-instance-groups --zone compute-zone resize my-cluster-minon-group --new-size 42
+```
+Instance Group will take care of putting appropriate image on new machines and start them, while Kubelet will register its Node with API server to make it available for scheduling. If you scale the instance group down, system will randomly choose Nodes to kill, so if you want to keep everything running all the time you probably shouldn't downsize the cluster twice.
 
-### Turn on or off an API version for your cluster
+In other environments you may need to configure the machine yourself and tell the Kubelet on which machine API server is running.
+TODO: flesh it out with some concrete example e.g. on Vagrant.
 
-Specific API versions can be turned on or off by passing --runtime-config=api/<version> flag while bringing up the server. For example: to turn off v1 API, pass --runtime-config=api/v1=false.
-runtime-config also supports 2 special keys: api/all and api/legacy to control all and legacy APIs respectively. For example, for turning off all api versions except v1, pass --runtime-config=api/all=false,api/v1=true.
+## Cluster introspection
```

Comment:
Actually, I would drop this whole section.
docs/user-guide/introspection-and-debugging.md
already talks about 'describe node' and
docs/user-guide/accessing-the-cluster.md
talks about 'cluster-info'
The one thing you might want to do is to add the info about 'get events' to
docs/user-guide/introspection-and-debugging.md